app/services/katello/event_queue.rb

```diff
@@ -0,0 +1,39 @@
+module Katello
+  class EventQueue
+    @event_types = {}
+
+    def self.clear_events(event_type, object_id, on_or_earlier_than)
+      Katello::Event.where(:in_progress => true, :object_id => object_id, :event_type => event_type).where('created_at <= ?', on_or_earlier_than).destroy_all
+    end
+
+    def self.next_event
+      first = Katello::Event.where(:in_progress => false).order(:created_at => 'asc').limit(1)
+      events = Katello::Event.where(:in_progress => false, :object_id => first.pluck(:object_id),
```

Comment:
If there are multiple events with the same object_id and event_type yes.  This could happen if for example two pulp applicability generation tasks finish within ~5 seconds of each other (or whatever the dynflow poll length is) and thus the applicability generation dynflow task sticks two on the queue.